### PR TITLE
Siadhandling

### PR DESCRIPTION
**Pull request by Mingling94**
`plugin.show()` and `plugin.hide()` now calls, via IPC, `view.start()` and `view.stop()` for semantics and to differentiate between the two. `plugin.show()` is called upon the button being pressed and incorporates transition logic, `view.start()` is called to start updating and api calling.

This PR makes siad startup handling a bunch more explicit, sending notifications to the user upon the daemon starting, stopping, and exiting. A new public boolean, named, "Running" in the DaemonManager keeps track of if the DaemonManager is aware that siad has started. Prevents an infinite flow of failed API calls when Running is set to false.

TODO after PR accepted: prettify notifications and allow user to turn siad on and off

### CODE REVIEW

**Comment by Mingling94** [comment]
boop @lukechampine @DavidVorick

**Comment by lukechampine** [comment]
why would the user want to turn siad on and off?

File: app/plugins/Overview/index.html [hunk 1]
```diff
@@ -22,6 +23,10 @@
 			<div class="welcome">
 				<div class="large">Welcome to Sia</div>
 				<div class="small">A highly efficient decentralized storage network.</div>
+				<div class="button" id="stop-exit">
+					<i class="fa fa-sign-out"></i>
+					Stop siad & exit
```
Comment (lukechampine) [inline app/plugins/Overview/index.html]: needs proper escaping (`&amp;`)

**Comment by Mingling94** [comment]
@lukechampine This only lets the user turn siad off. The UI turns siad on startup. @DavidVorick suggested that there should be a siad on/off button but it is an odd use case I suppose.

**Comment by Mingling94** [comment]
@lukechampine I took out the stop siad thing in overview as well as in daemonManager until these use cases become a bit more clear.

**Comment by lukechampine** [comment]
needs rebase

**Comment by Mingling94** [comment]
@lukechampine rebased

**Pull request merged**